neuralmonkey/runners/xent_runner.py

```diff
@@ -9,25 +9,26 @@
 from neuralmonkey.runners.base_runner import BaseRunner
 
 
-class PerplexityRunner(BaseRunner[AutoregressiveDecoder]):
+SupportedDecoders = Union[AutoregressiveDecoder]
+
+
+class XentRunner(BaseRunner[SupportedDecoders]):
```

Comment:
podle mě neni dovolený odvozovat generic od unionu. mypy nenadává? v pythonu 3.7 se budou dát vytasit protokoly a tady říct, že je to odvozený od čehokoliv, co má `train_xents`.

Reply:
A nebo se muze vykuchat spolecny predek/interface ktery bude podporovat cost, train_xents apod.

Reply:
To bych nedělal, právě proto, že od toho budou v Pythonu ty protokoly. Přehnaná dědičnost není pythonic.. pokud to projde přes mypy s unionem, tak bych to neměnil